### ISSUE TITLE
sidecar: broken with prometheus 2.7.1

### ISSUE THREAD

**Issue by tik-stbuehler** [thanos-io/thanos#4349]
**Thanos, Prometheus and Golang version used**:

prometheus from debian buster (Built-Using: golang-1.11), thanos binary from github release.

```
prometheus, version 2.7.1+ds (branch: debian/sid, revision: 2.7.1+ds-3+b11)
  build user:       pkg-go-maintainers@lists.alioth.debian.org
  build date:       20190608-09:54:04
  go version:       go1.11.6

thanos, version 0.21.1 (branch: HEAD, revision: 3558f4a5adc8d4068f82528570a439d440f98b9d)
  build user:       root@744cf7ef4576
  build date:       20210604-12:11:58
  go version:       go1.16.5
  platform:         linux/amd64
```

**What happened**:

sidecar can't connect to prometheus.

**Full logs to relevant components**:

Lots of these:
```
thanos[10952]: level=warn ts=2021-06-17T07:46:48.375156628Z caller=sidecar.go:159 msg="failed to fetch prometheus version. Is Prometheus running? Retrying" err="expected 2xx response, got 405. Body: Method Not Allowed\n"
```

**Anything else we need to know**:

Contrary to what the code comment in https://github.com/thanos-io/thanos/blob/7aa6f9760f0ab8617b356a6f6d5a98c7d361ab3e/pkg/promclient/promclient.go#L622-L629 says, older prometheus versions not always return 404 for not support API endpoints:

```
# curl -D- http://localhost:9090/api/v1/status/buildinfo
HTTP/1.1 405 Method Not Allowed
Content-Type: text/plain; charset=utf-8
X-Content-Type-Options: nosniff
Date: Thu, 17 Jun 2021 07:47:58 GMT
Content-Length: 19

Method Not Allowed
```

**Comment by GiedriusS** [comment]
So, it sounds like it should be enough to fix that `if`? BTW, why are you running such an old version? Is there any particular reason(-s)?

**Comment by tik-stbuehler** [comment]
I didn't test it, but yes, I'd guess that should be good enough. Perhaps consider all 4xx responses as "prometheus too old for this" - during startup I think I saw 5xx codes.

debian buster is still the current stable release. Running servers on anything but a stable release (with proper security support) is imho a real bad idea.

As thanos isn't part of debian we pick latest upstream (wrapping it in a local package) - not because I want to run latest, but because there is no (sane) other way.

Given bullseye is almost stable I now moved our prometheus/thanos setup to bullseye; so I'm not affected by this issue anymore.

(Fixing this issue might as well be "raise minimum required prometheus version"; but right now the docs say `v2.2.1+`.)
